Fix VB IUsingStatement.Declaration to not have null syntax.

333fred opened the pull request:
**Customer scenario**

When using VB `Using` statements with the FxCop analyzers, vbc will crash with:
```
Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalysisContextInfo.AnalysisContextInfo(Microsoft.CodeAnalysis.Compilation compilation, Microsoft.CodeAnalysis.IOperation operation) Line 45	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteOperationAction(Microsoft.CodeAnalysis.Diagnostics.OperationAnalyzerAction operationAction, Microsoft.CodeAnalysis.IOperation operation, Microsoft.CodeAnalysis.ISymbol containingSymbol, Microsoft.CodeAnalysis.SemanticModel semanticModel, System.Action<Microsoft.CodeAnalysis.Diagnostic> addDiagnostic, System.Func<Microsoft.CodeAnalysis.Diagnostic, bool> isSupportedDiagnostic, Microsoft.CodeAnalysis.Diagnostics.AnalysisState.OperationAnalyzerStateData analyzerStateOpt) Line 583	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteOperationActions(Microsoft.CodeAnalysis.IOperation operation, System.Collections.Generic.IDictionary<Microsoft.CodeAnalysis.OperationKind, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Diagnostics.OperationAnalyzerAction>> operationActionsByKind, Microsoft.CodeAnalysis.ISymbol containingSymbol, Microsoft.CodeAnalysis.SemanticModel model, System.Action<Microsoft.CodeAnalysis.Diagnostic> addDiagnostic, System.Func<Microsoft.CodeAnalysis.Diagnostic, bool> isSupportedDiagnostic, Microsoft.CodeAnalysis.Diagnostics.AnalysisState.OperationAnalyzerStateData analyzerStateOpt) Line 1147	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteOperationActions(System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.IOperation> operationsToAnalyze, System.Collections.Generic.IDictionary<Microsoft.CodeAnalysis.OperationKind, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Diagnostics.OperationAnalyzerAction>> operationActionsByKind, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer analyzer, Microsoft.CodeAnalysis.ISymbol containingSymbol, Microsoft.CodeAnalysis.SemanticModel model, System.Action<Microsoft.CodeAnalysis.Diagnostic> addDiagnostic, System.Func<Microsoft.CodeAnalysis.Diagnostic, bool> isSupportedDiagnostic, Microsoft.CodeAnalysis.Diagnostics.AnalysisState.OperationAnalyzerStateData analyzerStateOpt) Line 1128	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteBlockActionsCore<Microsoft.CodeAnalysis.Diagnostics.OperationBlockStartAnalyzerAction, Microsoft.CodeAnalysis.Diagnostics.OperationBlockAnalyzerAction, Microsoft.CodeAnalysis.Diagnostics.OperationAnalyzerAction, Microsoft.CodeAnalysis.Diagnostics.AnalysisState.OperationAnalyzerStateData, Microsoft.CodeAnalysis.IOperation, int>(System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.OperationBlockStartAnalyzerAction> startActions, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.OperationBlockAnalyzerAction> actions, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.OperationBlockAnalyzerAction> endActions, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer analyzer, Microsoft.CodeAnalysis.SyntaxNode declaredNode, Microsoft.CodeAnalysis.ISymbol declaredSymbol, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.IOperation> executableBlocks, System.Func<System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.IOperation>, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.IOperation>> getNodesToAnalyze, Microsoft.CodeAnalysis.SemanticModel semanticModel, System.Func<Microsoft.CodeAnalysis.SyntaxNode, int> getKind, Microsoft.CodeAnalysis.Diagnostics.AnalysisState.BlockAnalyzerStateData<Microsoft.CodeAnalysis.Diagnostics.OperationBlockAnalyzerAction, Microsoft.CodeAnalysis.Diagnostics.AnalysisState.OperationAnalyzerStateData> analyzerStateOpt, bool isGeneratedCode) Line 817	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.TryExecuteOperationBlockActions(System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.OperationBlockStartAnalyzerAction> operationBlockStartActions, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.OperationBlockAnalyzerAction> operationBlockActions, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.OperationBlockAnalyzerAction> operationBlockEndActions, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer analyzer, Microsoft.CodeAnalysis.SyntaxNode declaredNode, Microsoft.CodeAnalysis.ISymbol declaredSymbol, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.IOperation> operationBlocks, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.IOperation> operations, Microsoft.CodeAnalysis.SemanticModel semanticModel, Microsoft.CodeAnalysis.SyntaxReference declaration, int declarationIndex, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope analysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState analysisStateOpt, bool isGeneratedCode) Line 667	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver<Microsoft.CodeAnalysis.VisualBasic.SyntaxKind>.TryExecuteDeclaringReferenceActions(Microsoft.CodeAnalysis.SyntaxReference decl, int declarationIndex, Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent symbolEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope analysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState analysisStateOpt, bool shouldExecuteSyntaxNodeActions, bool shouldExecuteOperationActions, bool shouldExecuteCodeBlockActions, bool shouldExecuteOperationBlockActions, bool isInGeneratedCode, System.Threading.CancellationToken cancellationToken) Line 1849	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver<Microsoft.CodeAnalysis.VisualBasic.SyntaxKind>.TryExecuteDeclaringReferenceActions(Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent symbolEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope analysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState analysisStateOpt, bool isGeneratedCodeSymbol, System.Threading.CancellationToken cancellationToken) Line 1639	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.TryProcessSymbolDeclared(Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent symbolEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope analysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState analysisStateOpt, System.Threading.CancellationToken cancellationToken) Line 964	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.TryProcessEventCore(Microsoft.CodeAnalysis.Diagnostics.CompilationEvent e, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope analysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState analysisStateOpt, System.Threading.CancellationToken cancellationToken) Line 919	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEvent(Microsoft.CodeAnalysis.Diagnostics.CompilationEvent e, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope analysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState analysisStateOpt, System.Threading.CancellationToken cancellationToken) Line 906	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessCompilationEventsCoreAsync(Microsoft.CodeAnalysis.Diagnostics.AnalysisScope analysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState analysisStateOpt, bool prePopulatedEventQueue, System.Threading.CancellationToken cancellationToken) Line 893	C#
```

**Bugs this fixes:**

Fixes https://github.com/dotnet/roslyn/issues/19819, https://github.com/dotnet/roslyn/issues/19887.

**Workarounds, if any**

Don't use the new IOperation-based analyzers, or don't use `Using` statements. There is no other workaround.

**Risk**

Should be low.

**Performance impact**

Low. No noticeable impact unless IOperation is used, and then we just have an additional reference to a SyntaxNode.

**Is this a regression from a previous update?**

No

**Root cause analysis:**

Insufficient dogfooding of our analyzers. This pr will unblock https://github.com/dotnet/roslyn/pull/19806, which will move us forward on that.

**How was the bug found?**

Internal PR blocker.

333fred commented:
Tagging @dotnet/analyzer-ioperation @mavasani for first pass review.

genlu commented on src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb:
```diff
@@ -352,5 +352,64 @@ BC30581: 'AddressOf' expression cannot be converted to 'Integer' because 'Intege
+Imports System
+Module Module1
+    Class C1
+        Inherits IDisposable
```
Implements?

333fred replied:
Interesting that the test didn't fail to compile...

333fred replied:
Fixed.

genlu commented on src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb:
```diff
@@ -352,5 +352,64 @@ BC30581: 'AddressOf' expression cannot be converted to 'Integer' because 'Intege
+            Dim op = DirectCast(comp.GetSemanticModel(tree).GetOperationInternal(node), IUsingStatement)
+
+            Assert.NotNull(op.Declaration.Syntax)
+            Assert.Equal("Using D1 as New C1()", op.Declaration.Syntax.ToString())
```
Consider sticking to `VerifyOperationTreeAndDiagnosticsForTest`, so we can check both the value of syntax and the shape of operation, plus this would catch any compilation error.

333fred replied:
Unfortunately we can't use that at the moment, because of https://github.com/dotnet/roslyn/issues/17917. I would've used it if I could, but VerifyOperation currently throws exceptions.

mavasani commented:
Also tagging @dotnet/roslyn-compiler for review.

genlu commented on src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb:
```diff
@@ -1460,7 +1461,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Function(BoundUsing)
-                        Return New Variables(BoundUsing.ResourceList.As(Of IVariableDeclaration))
+                        Dim usingStatementSyntax = Syntax.ChildNodes().OfType(Of UsingStatementSyntax).SingleOrDefault()
+                        Return New Variables(BoundUsing.ResourceList.As(Of IVariableDeclaration), usingStatementSyntax)
```
This might be an existing issue, the comment on the interface says
```cs
        /// <summary>
        /// Declaration introduced by the using statement. Null if the using statement does not declare any variables.
        /// </summary>
        IVariableDeclarationStatement Declaration { get; }
```
But here it seems we always return an operation for declaration.

genlu replied:
Also, could you please add a test for this scenario?
```vb
Imports System
Module Module1
    Class C1
        Implements IDisposable

        Public Sub Dispose() Implements IDisposable.Dispose
            Throw New NotImplementedException()
        End Sub
    End Class
    Sub S1()
        Dim x = New C1()
        Using x
        End Using
    End Sub
End Module
```

333fred replied:
Done.

AlekseyTs replied:
>Return New Variables(BoundUsing.ResourceList.As(Of IVariableDeclaration), usingStatementSyntax) [](start = 24, length = 95)

We are fixing null syntax, but it looks like IVariableDeclarationStatement.Declarations can return default array for this object. Are we fine leaving product in this state? It seems addressing this issue is as easy as checking if BoundUsing.ResourceList is default.

AlekseyTs replied:
Of course, there is a risk that there is a consumer that doesn't handle null from IUsingStatement.Declaration.

---
In reply to: [119527163](https://github.com/dotnet/roslyn/pull/19885#discussion_r119527163) [](ancestors = 119527163)

AlekseyTs replied:
However, it looks like C# implementation already returns null from this API.

---
In reply to: [119527306](https://github.com/dotnet/roslyn/pull/19885#discussion_r119527306) [](ancestors = 119527306,119527163)

AlekseyTs replied:
However, it looks like C# implementation already returns null from this API.

333fred replied:
It seems to be a fairly easy change to make it conform to the interface docs, so I'm going with that route.

genlu commented on src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb:
```diff
@@ -352,5 +352,70 @@ BC30581: 'AddressOf' expression cannot be converted to 'Integer' because 'Intege
+            Dim op = DirectCast(comp.GetSemanticModel(tree).GetOperationInternal(node), IUsingStatement)
+
+            Assert.NotNull(op.Declaration.Syntax)
+            Assert.Equal("Using", op.Declaration.Syntax.ToString())
```
As mentioned above, do we want to return an non-null operation for `Declaration` in this case?

333fred replied:
I'm thinking that's a separate issue, this PR isn't touching the declaration, just the syntax node of the variable returned.

333fred replied:
Created https://github.com/dotnet/roslyn/issues/19887 for the issue.

333fred commented:
Tagging @dotnet/roslyn-compiler @AlekseyTs; can I get a couple of reviews please?

jcouv commented on src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb:
```diff
@@ -1487,9 +1489,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
+            Private ReadOnly _syntax As SyntaxNode
 
-            Public Sub New(variables As ImmutableArray(Of IVariableDeclaration))
+            Public Sub New(variables As ImmutableArray(Of IVariableDeclaration), syntax As SyntaxNode)
```
Could the type of syntax be more specific (`UsingStatementSyntax`)?

333fred replied:
Done.

jcouv replied:
Thanks. Then it seems `_syntax` can also use the specific type. And the cast above may become irrelevant.

---
In reply to: [119499384](https://github.com/dotnet/roslyn/pull/19885#discussion_r119499384) [](ancestors = 119499384)

jcouv commented on src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb:
```diff
@@ -352,5 +352,103 @@ BC30581: 'AddressOf' expression cannot be converted to 'Integer' because 'Intege
+
+        <Fact>
+        Public Sub UsingDeclarationExistingVariableNonNullSyntax()
+            ' TODO: This will need to be removed when https://github.com/dotnet/roslyn/issues/19887 is fixed.
```
Could you include a comment about actual vs expected behavior, in addition to the link to the issue?

333fred replied:
Updated.

333fred commented:
@dotnet/roslyn-compiler, could I get another review please?

333fred commented:
@dotnet/roslyn-infrastructure @jasonmalinowski test failure appears to be https://github.com/dotnet/roslyn/issues/14592.
```
 Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces.ProjectCacheHostServiceFactoryTests.TestImplicitCacheMonitoring
      Reference should have been released but was not.
      Expected: False
      Actual:   True
      Stack Trace:
           at Roslyn.Test.Utilities.ObjectReference`1.AssertReleased()
        q:\roslyn\src\EditorFeatures\Test\Workspaces\ProjectCacheHostServiceFactoryTests.cs(118,0): at Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces.ProjectCacheHostServiceFactoryTests.TestImplicitCacheMonitoring()
```

333fred commented:
@dotnet-bot retest windows_release_unit64_prtest please

jasonmalinowski commented:
@333fred Thanks for investigating before retesting!

AlekseyTs commented on src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb:
```diff
@@ -1460,7 +1461,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Me,
                     Function(BoundUsing)
-                        Return New Variables(BoundUsing.ResourceList.As(Of IVariableDeclaration))
+                        Dim usingStatementSyntax = Syntax.ChildNodes().OfType(Of UsingStatementSyntax).SingleOrDefault()
```
>Syntax.ChildNodes().OfType(Of UsingStatementSyntax).SingleOrDefault() [](start = 51, length = 69)

It looks like the syntax is always a UsingBlockSyntax, so this could be just ```DirectCast(Syntax, UsingBlockSyntax).UsingStatement```.

AlekseyTs replied:
We don't want this to be default ever, right?

---
In reply to: [119525947](https://github.com/dotnet/roslyn/pull/19885#discussion_r119525947) [](ancestors = 119525947)

AlekseyTs replied:
It doesn't look like this suggestion was considered for the last iteration.

333fred replied:
What do you mean? I'm now returning `Nothing` in the default case, as per the docs, and this should never be default

AlekseyTs commented on src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb:
```diff
@@ -352,5 +352,106 @@ BC30581: 'AddressOf' expression cannot be converted to 'Integer' because 'Intege
+            Dim op = DirectCast(comp.GetSemanticModel(tree).GetOperationInternal(node), IUsingStatement)
+
+            Assert.NotNull(op.Declaration.Syntax)
+            Assert.Equal("Using D1 as New C1()", op.Declaration.Syntax.ToString())
```
>Assert.Equal("Using D1 as New C1()", op.Declaration.Syntax.ToString()) [](start = 12, length = 70)

```Assert.Same(node.UsingStatement, op.Declaration.Syntax)```?

333fred replied:
Done

AlekseyTs commented on src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb:
```diff
@@ -352,5 +352,106 @@ BC30581: 'AddressOf' expression cannot be converted to 'Integer' because 'Intege
+
+        <Fact>
+        <WorkItem(19819, "https://github.com/dotnet/roslyn/issues/19819")>
+        Public Sub UsingDeclarationSyntaxNotNull()
```
>UsingDeclarationSyntaxNotNull [](start = 19, length = 29)

Please assert diagnostics in all new tests, we should see if there is anything wrong with the code and what exactly is wrong.

333fred replied:
Done

AlekseyTs commented on src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb:
```diff
@@ -352,5 +352,106 @@ BC30581: 'AddressOf' expression cannot be converted to 'Integer' because 'Intege
+            Dim op = DirectCast(comp.GetSemanticModel(tree).GetOperationInternal(node), IUsingStatement)
+
+            Assert.NotNull(op.Declaration.Syntax)
+            Assert.Equal("Using", op.Declaration.Syntax.ToString())
```
>Assert.Equal("Using", op.Declaration.Syntax.ToString()) [](start = 12, length = 55)

Same suggestion as for the test above.

333fred replied:
No longer relevant, as this Declaration is now null as per docs.

AlekseyTs commented on src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb:
```diff
@@ -352,5 +352,106 @@ BC30581: 'AddressOf' expression cannot be converted to 'Integer' because 'Intege
+            Dim op = DirectCast(comp.GetSemanticModel(tree).GetOperationInternal(node), IUsingStatement)
+
+            Assert.NotNull(op.Declaration.Syntax)
+            Assert.Equal("Using x", op.Declaration.Syntax.ToString())
```
>Assert.Equal("Using x", op.Declaration.Syntax.ToString()) [](start = 12, length = 57)

Same suggestion as for the test above.

333fred replied:
No longer relevant, as this Declaration is now null as per docs.

AlekseyTs commented:
Done with review pass.

333fred commented:
@AlekseyTs addressed feedback. We now return a `Nothing` declaration array when the underlying `ResourceList` is default. From my reading of the binder, there shouldn't be a scenario where we have a non-default, empty `ResourceList`.

AlekseyTs commented on src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb:
```diff
@@ -1460,7 +1461,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
+                        If (BoundUsing.ResourceList.IsDefault) Then
+                            Return Nothing
+                        End If
+                        Dim usingStatementSyntax = DirectCast(Syntax.ChildNodes()(0), UsingStatementSyntax)
```
>DirectCast(Syntax.ChildNodes()(0), UsingStatementSyntax) [](start = 51, length = 56)

Why not  ```DirectCast(Syntax, UsingBlockSyntax).UsingStatement```?

333fred replied:
Because I apparently misread your previous comment and it didn't occur to me :).

333fred replied:
Fixed

AlekseyTs commented on src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb:
```diff
@@ -1460,7 +1461,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Me,
                     Function(BoundUsing)
-                        Return New Variables(BoundUsing.ResourceList.As(Of IVariableDeclaration))
+                        If (BoundUsing.ResourceList.IsDefault) Then
```
>(BoundUsing.ResourceList.IsDefault) [](start = 27, length = 35)

Minor point, parenthesis around condition are not required in VB.

333fred replied:
Fixed

AlekseyTs commented on src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb:
```diff
@@ -16,24 +16,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics
                                  <![CDATA[
 Public Class B2
-    Public Shared Operator +(x As B2, y As B2) As B2 
+    Public Shared Operator +(x As B2, y As B2) As B2
```
It looks like there are no meaningful changes in this file, consider reverting to the previous state.

333fred replied:
Reverted

333fred commented:
Tagging @MattGertz for ask mode approval.

jcouv commented on src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb:
```diff
@@ -1487,9 +1492,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Implements IVariableDeclarationStatement
 
             Private ReadOnly _variables As ImmutableArray(Of IVariableDeclaration)
+            Private ReadOnly _syntax As SyntaxNode
```
Can use the specific type here. Then remove the cast above (line 1467).

333fred replied:
No, that `Syntax` is from `BoundNode`, http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/BoundTree/BoundNode.cs,4494ab96c2bc2ab4. Making this more specific won't allow me to remove that cast.

333fred commented:
@dotnet-bot restest windows_debug_vs-integration_prtest please.

@jaredpar suggested that jenkins images were just updated, and it was rebooting.

333fred commented:
NuGet cache corruption

NuGet hit a corrupted cache on this machine. Please delete the agent and then retest, noting you hit "NuGet cache corruption with null files".

333fred commented:
@dotnet-bot retest windows_build_correctness_prtest please

333fred commented:
@dotnet-bot retest windows_debug_unit32_prtest please

333fred commented:
Ok, new round of failures. windows_release_unit64_prtest failed with:
```
11:54:38 Agent went offline during the build
11:54:38 
ERROR: Connection was broken: java.io.IOException: Connection aborted: org.jenkinsci.remoting.nio.NioChannelHub$MonoNioTransport@5214f8fa[name=win2016-5fe5c0]
11:54:38 	at org.jenkinsci.remoting.nio.NioChannelHub$NioTransport.abort(NioChannelHub.java:208)
11:54:38 	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:629)
11:54:38 	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
11:54:38 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
11:54:38 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
11:54:38 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
11:54:38 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
11:54:38 	at java.lang.Thread.run(Thread.java:748)
11:54:38 Caused by: java.io.IOException: Connection timed out
11:54:38 	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
11:54:38 	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
11:54:38 	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
11:54:38 	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
11:54:38 	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
11:54:38 	at org.jenkinsci.remoting.nio.FifoBuffer$Pointer.receive(FifoBuffer.java:137)
11:54:38 	at org.jenkinsci.remoting.nio.FifoBuffer.receive(FifoBuffer.java:310)
11:54:38 	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:561)
11:54:38 	... 6 more
```

333fred commented:
@dotnet-bot retest windows_release_unit64_prtest please

333fred commented:
Emailed @mmitche about the coreclr failure.

333fred commented:
@dotnet-bot retest windows_coreclr_test_prtest please

333fred commented:
@dotnet/roslyn-infrastructure, I'm unsure of what caused https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_unit64_prtest/7037/ to fail. The log says that there was a failure in the EditorServicesTests suite, but the xunit log has no failures.

333fred commented:
@dotnet-bot retest windows_debug_unit64_prtest please

333fred commented:
Debug integration failure was https://github.com/dotnet/roslyn/issues/19815

333fred commented:
@dotnet-bot retest windows_debug_vs-integration_prtest please
